header.go

```diff
@@ -54,11 +57,14 @@ func invoiceNumber(inv *bill.Invoice) string {
 // - 384 (Corrected invoice)
 // - 389 (Self-billed invoice)
 // - 381 (Credit note)
-func invoiceTypeCode(t cbc.Key) string {
+func invoiceTypeCode(inv *bill.Invoice) string {
+	if inv.Tax != nil && inv.Type == bill.InvoiceTypeStandard && slices.Contains(inv.Tax.Tags, tax.TagSelfBilled) {
```

Comment:
This condition looks complicated. What exactly does it do?
I have an idea to move this condition to private method. The method name could be descriptive and return true/false.

Reply:
For the invoice to be a self billed invoice (code 389) in gobl it has to be an invoice (inv.Type == bill.InvoiceTypeStandard) and it has to have the self billed tag (slices.Contains(inv.Tax.Tags, tax.TagSelfBilled)).
We also check here if the inv.Tax structure is present because if it isn't then checking the self billed tag would panic.

Reply:
What do you think about a function instead of a condition?
```go
func isSelfBilledInvoice(inv *bill.Invoice) bool {
  return inv.Tax != nil && inv.Type == bill.InvoiceTypeStandard && slices.Contains(inv.Tax.Tags, tax.TagSelfBilled)
}
```
Then we can:
```suggestion
	if isSelfBilledInvoice(inv) {
```